PR : 실습 2주차 과제

Jokwanhee opened the pull request:
## 📌𝘐𝘴𝘴𝘶𝘦𝘴
- closed #5
## 📎𝘞𝘰𝘳𝘬 𝘋𝘦𝘴𝘤𝘳𝘪𝘱𝘵𝘪𝘰𝘯
-  Fragment 생명주기 이해하기 (README.md 파일에 노션 링크 첨부)
- BottomNavigation 디자인 추가하기
  - color, ripple, selected ...
- DiffUtil & ListAdapter 사용하기

## 📷𝘚𝘤𝘳𝘦𝘦𝘯𝘴𝘩𝘰𝘵
https://github.com/DO-SOPT-ANDROID/kwanhee-jo/assets/90740783/02f3331b-96e4-44ea-b414-e2502e40d240

## 💬𝘛𝘰 𝘙𝘦𝘷𝘪𝘦𝘸𝘦𝘳𝘴
ListAdapter 를 사용하면 notifyDataSetChanged 의 의미를 알아볼 수 있다고 했던 리뷰가 기억이나는 과제였습니다! 역시 리뷰가 큰 힘이 되는 것 같아요! 리뷰는 항상 감사합니다 🦭

Jokwanhee committed:
[add] bottom navigation reselected

바텀네비게이션을 클릭 후 다시 클릭하면 나타나는 리스너로 recyclerview scroll position 을 상단으로 올리는 기능

Jokwanhee committed:
[add] bottom navigation ui

바텀 네비게이션 ui 및 기능 추가

Jokwanhee committed:
[add] migration RecyclerView to ListAdatper & DiffUtil

리사이클러뷰에서 DiffUtil 로 마이그레이션하기

Jokwanhee committed:
[add] week2 notion into README.md

2주차 심화 과제 노션 링크

emjayMJkim commented on app/src/main/java/org/sopt/dosopttemplate/presentation/home/HomeActivity.kt:
```diff
@@ -25,6 +29,17 @@ class HomeActivity : BaseActivity<ActivityHomeBinding>(ActivityHomeBinding::infl
+        initBottomNavigationReSelected()
+    }
+
+    private fun initBottomNavigationReSelected() {
```
이 함수를 만들어준 이유가 뭔가요??

Jokwanhee replied:
바텀 네비게이션을 클릭했을 때, 스크롤을 위로 올리기위한 함수입니다!

emjayMJkim commented on app/src/main/java/org/sopt/dosopttemplate/adapter/HomeAdapter.kt:
```diff
@@ -2,15 +2,16 @@ package org.sopt.dosopttemplate.adapter
 
-class HomeAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
-    private var profileList = emptyList<HomeProfileModel>()
+class HomeAdapter : ListAdapter<HomeProfileModel, RecyclerView.ViewHolder>(diffUtil) {
```
listadapter 👍  저도 빨리 listadapter 적용할께요 ㅎㅎ

emjayMJkim reviewed (commented):
역시 믿고 보는 코드..!! 수고했어!!

Jokwanhee committed:
[add] landscape screen

가로모드 대응화면 만들기(viewPage2 & TabLayout) 사용
별도로 가로모드 Adapter 클래스 작성

Jokwanhee committed:
[add] adapter item click listener

클릭 이벤트 추가

Jokwanhee committed:
[add] bind HomeDetailActivity UI

상세 보기 화면 UI 바인딩

Jokwanhee committed:
[add] HomeDetail birthday & music UI

생일 & 뮤직 ui 추가

Jokwanhee committed:
[add] HomeDetail back button

세부 사항 액티비티 뒤로가기 버튼 만들기

Jokwanhee committed:
[add] add item & remove item

아이템 삭제 및 추가 (다이어로그 사용)
친구 생성 시 생일 확인 후 생일 프로필 생성
가로모드 대응
삭제 및 추가 시 리스트 갱신
친구 삭제 시 생일 + 기존 아이템 중복 삭제

Jokwanhee committed:
[review] 의진 리뷰

프로필 데이터를 생일 데이터로 매핑하는 확장함수 만들기

Jokwanhee committed:
[review] 의진 리뷰 피드백

어뎁터 사용으로 메모리 누수 방지
- RecyclerView adapter
- ViewPager2 adapter

Jokwanhee committed:
[del] Extensions func about Glide

쓸모없는 Glide 확장 함수 코드 삭제

Jokwanhee committed:
Merge pull request #8 from DO-SOPT-ANDROID/feat/week2_challenge

PR : 실습 2주차 과제
